clang/include/clang/AST/Stmt.h

```diff
@@ -2080,6 +2080,7 @@ class IfStmt final
   /// If this is an 'if constexpr', determine which substatement will be taken.
   /// Otherwise, or if the condition is value-dependent, returns None.
   Optional<const Stmt*> getNondiscardedCase(const ASTContext &Ctx) const;
```

Comment:
The usual approach for this sort of dance is to have the non-`const` implementation be the "real" implementation, and have the `const` one use `const_cast` to call into the non-`const` one. This way we don't have to duplicate the logic, and given that there's no way to get a truly `const` AST node (they're all heap allocated and thus never declared as `const` when initially created), there's no UB to worry about. WDYT?

Reply:
The reason I didn't was that we end up changing return value, and getting the 'type' re-assembled out of the Optional ends up being pretty awkward.  It isn't just changing the return value, it is creating a new 'Optional' type that doesn't match.

I can give it a shot if you're attached to it.

Reply:
Nope, not that attached to the idea.